packages/flutter_tools/test/general.shard/base/error_handling_io_test.dart

```diff
@@ -372,58 +384,131 @@ void main() {
+      final File file = directory.childFile('file');
 
-      final File file = fs.file('file');
-
-      const String expectedMessage = 'The flutter tool cannot access the file or directory';
-      expect(() async => file.writeAsBytes(<int>[0]),
-             throwsToolExit(message: expectedMessage));
-      expect(() async => file.writeAsString(''),
-             throwsToolExit(message: expectedMessage));
-      expect(() => file.writeAsBytesSync(<int>[0]),
-             throwsToolExit(message: expectedMessage));
-      expect(() => file.writeAsStringSync(''),
-             throwsToolExit(message: expectedMessage));
-      expect(() => file.openSync(),
-             throwsToolExit(message: expectedMessage));
-      expect(() => file.createSync(),
-             throwsToolExit(message: expectedMessage));
+      exceptionHandler.addError(
+        file,
+        FileSystemOp.create,
+        FileSystemException('', file.path, const OSError('', eacces)),
+      );
+      exceptionHandler.addError(
+        file,
+        FileSystemOp.write,
+        FileSystemException('', file.path, const OSError('', eacces)),
+      );
+      exceptionHandler.addError(
+        file,
+        FileSystemOp.read,
+        FileSystemException('', file.path, const OSError('', eacces)),
+      );
+      exceptionHandler.addError(
+        file,
+        FileSystemOp.delete,
+        FileSystemException('', file.path, const OSError('', eacces)),
+      );
+      const String writeMessage =
+          'Flutter failed to write to a file at "dir/file". The flutter tool cannot access the file or directory.\n'
+          'Please ensure that the SDK and/or project is installed in a location that has read/write permissions for the current user.\n'
+          'Try running:\n'
+          r'  sudo chown -R $(whoami) /dir/file';
+      expect(() async => file.writeAsBytes(<int>[0]), throwsToolExit(message: writeMessage));
+      expect(() async => file.writeAsString(''), throwsToolExit(message: writeMessage));
+      expect(() => file.writeAsBytesSync(<int>[0]), throwsToolExit(message: writeMessage));
+      expect(() => file.writeAsStringSync(''), throwsToolExit(message: writeMessage));
+
+      const String createMessage =
+          'Flutter failed to create file at "dir/file". The flutter tool cannot access the file or directory.\n'
+          'Please ensure that the SDK and/or project is installed in a location that has read/write permissions for the current user.\n'
+          'Try running:\n'
+          r'  sudo chown -R $(whoami) /dir';
+      expect(() => file.createSync(), throwsToolExit(message: createMessage));
+      try {
+        file.createSync(recursive: true);
+        expect(false, isTrue); // Shouldn't reach here.
+      } on ToolExit catch (error) {
+        // Recursive does not contain the "sudo chown" suggestion.
+        expect(error.message, isNot(contains('sudo chown')));
+      }
```

Comment:
I couldn't get negative lookaheads to work with `RegExp` to pass into the `throwsToolExit` message.  Instead of trying to be cute, just catch it and check more explicitly.

Reply:
A slightly better pattern might be:

```dart

expect(() => ...., throwsA(isA<ToolExit>().having((e) => e.message, 'message', isNot(...))))
```

Reply:
Good call, that's way better.